examples/mongocxx/explicit_encryption_auto_decryption.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+#include <iostream>
+#include <random>
+
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/builder/basic/kvp.hpp>
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/builder/stream/helpers.hpp>
+#include <bsoncxx/json.hpp>
+#include <bsoncxx/types/bson_value/make_value.hpp>
+#include <mongocxx/client.hpp>
+#include <mongocxx/client_encryption.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/options/auto_encryption.hpp>
+#include <mongocxx/options/client.hpp>
+#include <mongocxx/options/data_key.hpp>
+#include <mongocxx/options/encrypt.hpp>
+
+using bsoncxx::builder::basic::make_document;
+using bsoncxx::builder::basic::kvp;
+
+using bsoncxx::builder::stream::document;
+using bsoncxx::builder::stream::open_document;
+using bsoncxx::builder::stream::close_document;
+using bsoncxx::builder::stream::finalize;
+
+using bsoncxx::types::bson_value::make_value;
+
+using namespace mongocxx;
+
+int main(int, char**) {
+    instance inst{};
+
+    // This must be the same master key that was used to create
+    // the encryption key; here, we use a random key as a placeholder.
+    auto key_length = 96;
+    char key_storage[96];
```

Comment:
Same comment about using a const / macro for 96.

Reply:
Done